x/ibc/02-client/keeper/keeper.go

```diff
@@ -206,6 +206,23 @@ func (k Keeper) GetSelfConsensusState(ctx sdk.Context, height uint64) (exported.
 	return consensusState, true
 }
 
+// GetSelfConsensusStateWithParams introspects the (self) past historical info at a given height
+// along with the appropriate client paramaters and returns the expected consensus state at that height
+func (k Keeper) GetSelfConsensusStateWithParams(ctx sdk.Context, height uint64) (exported.ConsensusState, bool) {
+	consState, ok := k.GetSelfConsensusState(ctx, height)
+	if !ok {
+		return nil, false
+	}
+
+	tmConsState := consState.(ibctmtypes.ConsensusState)
+	// inject introspected parameters
+	tmConsState.ConsensusParams = ctx.ConsensusParams()
```

Comment:
Need to see if there's a way to get these paramaters at the given height, rather than just latest